.github/workflows/tekton_bundle_push.yaml

```diff
@@ -17,21 +15,58 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: debianmaster/actions-k3s@master
-        id: k3s
-        with:
-          version: 'latest'
+      - run: |
+          mkdir -vp $HOME/{bin,.kube} || true
+          pushd $HOME/bin
+          curl -LO https://dl.k8s.io/release/v1.23.0/bin/linux/amd64/kubectl
+          popd
+          PATH="$HOME/bin:$PATH"
+          cat <<-EOF > $HOME/.kube/config
+          apiVersion: v1
+          clusters:
+          - cluster:
+              insecure-skip-tls-verify: true
+              server: https://api.example.com:6443
+            name: api-example-dot-com:6443
+          contexts:
+          - context:
+              cluster: api-example-dot-com:6443
+              namespace: default
+              user: kube:admin/api-example-dot-com:6443
+            name: default/api-example-dot-com:6443/kube:admin
+          current-context: default/api-example-dot-com:6443/kube:admin
+          kind: Config
+          EOF
       - uses: jerop/tkn@v0.1.0
       - uses: redhat-actions/podman-login@v1
         with:
           username: ${{ env.REGISTRY_USER }}
           password: ${{ env.REGISTRY_PASSWORD }}
           registry: ${{ env.IMAGE_REGISTRY }}
       - run: |
-          printf -v image_string '%s/%s:%s' \
-            ${{ env.IMAGE_REGISTRY }} \
-            ${{ env.IMAGE_NAMESPACE }} \
-            ${{ env.IMAGE_TAG }} ;
-          printf -v bundle_path '%s/%s' $GITHUB_WORKSPACE ${{ env.BUNDLE }} ;
-          tkn bundle push -f ${bundle_path} ${image_string} ;
-          tkn bundle list -o yaml --no-cache ${image_string} ;
\ No newline at end of file
+          for dir in ${GITHUB_WORKSPACE}/*
+          do
+            [[ -d "$dir" ]] || continue
+            printf 'Found Bundle dir: %s\n' $dir
+            BUNDLE_NAME="${dir##*/}"
+            BUNDLE_FILES=()
+            for file in  $dir/*.y{,a}ml
+            do
+              [[ -e "$file" ]] || continue
+              printf " * %s\n" $file
+              BUNDLE_FILES+=("$file")
+            done
+            printf -v bundle_files -- '-f %s ' ${BUNDLE_FILES[*]}
+            printf -v image_sha '%s/%s/%s:%s' \
+              ${{ env.IMAGE_REGISTRY }} \
+              ${{ env.IMAGE_NAMESPACE }} \
+              "${BUNDLE_NAME}" \
+              "${GITHUB_SHA:0:8}" ;
+            printf -v image_latest '%s/%s/%s:%s' \
+              ${{ env.IMAGE_REGISTRY }} \
+              ${{ env.IMAGE_NAMESPACE }} \
+              "${BUNDLE_NAME}" \
+              "latest" ;
+            tkn bundle push ${bundle_files} $image_sha ;
+            tkn bundle push ${bundle_files} $image_latest ;
+          done
```

Comment:
This might be a dupe comment, but I'll respond anyways...

I do not like this code, it's very simple, and has zero nuance, but it does the job in a brutal yet effective way.
We want a more elegant solution, and that's being worked on.

So I've been thinking about the last part, about Quay repos.

I've been looking at the Quay API, and need to figure out how to do the following:
* create a new repo when a new bundle appears here on GitHub.
  * new Quay repo will take the name of the bundle folder.
* add an existing robot account to the new Quay repo.

I could write a quick python script probably within a few hours, but I'm more interested in golang. I may prototype in python, and move to golang... or, we can decide to create Quay repos manually for the time being?

Reply:
> Problem with this is that, if I understood everything correctly, any change in the repo will make the CI to push a bundle per directory (with or without changes in that particular directory). Not what we want. New tags should be pushed only for directories with changes.

That is correct, and I agree that's an issue. 
I'll have to figure out a way to see what files are changing, and replace this code.

Reply:
I don't think you need python nor go. You just need to make a query with curl to the Quay API to create a repo (you need credentials) with a public visibility, if it doesn't exist already. Then you use your very same code to push bundles. Something on that line.

Reply:
note to self: https://docs.quay.io/api/swagger/

Reply:
fixed